__integrations__/__tests__/getAssertion.test.ts

```diff
@@ -0,0 +1,357 @@
+
+import {
+  createCosmosDbAndCollections,
+  LOLLIPOP_COSMOSDB_COLLECTION_NAME
+} from "../utils/fixtures";
+import { createBlobs } from "../utils/azure_storage";
+import {
+  fetchActivatePubKey,
+  fetchGenerateLcParams,
+  fetchGetAssertion,
+  fetchReservePubKey
+} from "../utils/client";
+
+import {
+  aFiscalCode,
+  aValidSha256AssertionRef,
+  aValidSha512AssertionRef
+} from "../../__mocks__/lollipopPubKey.mock";
+
+const MAX_ATTEMPT = 50;
+
+jest.setTimeout(WAIT_MS * MAX_ATTEMPT);
+
+const customHeaders = {
+  "x-user-groups": "ApiLollipopAssertionRead",
+  "x-subscription-id": "anEnabledServiceId",
+  "x-user-email": "unused@example.com",
+  "x-user-id": "unused",
+  "x-user-note": "unused",
+  "x-functions-key": "unused",
+  "x-forwarded-for": "0.0.0.0",
+  "Ocp-Apim-Subscription-Key": "aSubscriptionKey"
+};
+
+const baseUrl = "http://function:7071";
+const myFetch = (getNodeFetch(customHeaders) as unknown) as typeof fetch;
+
+const LOLLIPOP_ASSERTION_STORAGE_CONTAINER_NAME = "assertions";
+
+// ----------------
+// Setup dbs
+// ----------------
+
+const blobService = createBlobService(QueueStorageConnection);
+
+// @ts-ignore
+const cosmosClient = new CosmosClient({
+  endpoint: COSMOSDB_URI,
+  key: COSMOSDB_KEY
+});
+
+// Wait some time
+beforeAll(async () => {
+  await pipe(
+    createCosmosDbAndCollections(COSMOSDB_NAME),
+    TE.getOrElse(() => {
+      throw Error("Cannot create infra resources");
+    })
+  )();
+  await pipe(
+    createBlobs(blobService, [LOLLIPOP_ASSERTION_STORAGE_CONTAINER_NAME]),
+    TE.getOrElse(() => {
+      throw Error("Cannot create azure storage");
+    })
+  )();
+
+  await waitFunctionToSetup();
+});
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const cosmosInstance = cosmosClient.database(COSMOSDB_NAME);
+const container = cosmosInstance.container(LOLLIPOP_COSMOSDB_COLLECTION_NAME);
+const lolliPOPKeysModel = new LolliPOPKeysModel(container);
+
+const aGenerateLcParamsPayload = {
+  operation_id: "an_operation_id" as NonEmptyString
+};
+
+const expires = date_fns.addDays(new Date(), 30);
+
+const validActivatePubKeyPayload: ActivatePubKeyPayload = {
+  assertion_type: AssertionTypeEnum.SAML,
+  assertion: "aValidAssertion" as NonEmptyString,
+  expired_at: expires,
+  fiscal_code: aFiscalCode
+};
+
+// this method generates new JWK for use in the describe below
+const generateJwkForTest = async (): Promise<JwkPublicKey> => {
+  const keyPair = await jose.generateKeyPair("ES256");
+  return (await jose.exportJWK(keyPair.publicKey)) as JwkPublicKey;
+};
+
+const generateAssertionRefForTest = async (
+  jwk: JwkPublicKey,
+  algo: JwkPubKeyHashAlgorithmEnum = JwkPubKeyHashAlgorithmEnum.sha256
+): Promise<AssertionRef> => {
+  const thumbprint = await jose.calculateJwkThumbprint(jwk, algo);
+  return `${algo}-${thumbprint}` as AssertionRef;
+};
+
+// -------------------------
+// Tests
+// -------------------------
+
+describe("getAssertion |> Validation Failures", () => {
```

Comment:
can we add a describe for the error `410` when a valid payload is sent but the `popDocument` or `assertion` are not found?

Reply:
Done in [437edfd](https://github.com/pagopa/io-functions-lollipop/pull/26/commits/437edfd1a5af541b2131040944239645491b3a3a)